src/bindgen/bindings.rs

```diff
@@ -85,6 +89,25 @@ impl Bindings {
         }
     }
 
+    pub fn embed_file<P: AsRef<path::Path>, F: Write>(&self, out: &mut SourceWriter<F>, path: P) {
+        let mut src_content = String::new();
+
+        let src_file = self
+            .root
+            .as_ref()
+            .expect("Bindings::root must be set when embedding files.")
```

Comment:
I don't know, using cwd by default, or config location otherwise doesn't seem too bad.

If I specify the path as an absolute file, which is more than likely when ran from build.rs, it doesn't seem necessary to have to provide a root path... But I'm very bad at API design myself, and this seems ok I guess.

Maybe @eqrion can take a look and comment on this?

Reply:
I feel strongly that the directory hosting the used `cbindgen.toml` must be the relative anchor for embeds, as that's the only thing you can rely on without thinking too much. Any working dir is random, and the `Cargo.toml` can get mildly ambiguous when generating bindings for a sub-project.

Reply:
I think having these paths be relative to 'cbindgen.toml' makes sense. I'm not great at API design either, so it's not a strong feeling.